megamek/src/megamek/common/QuadMech.java

```diff
@@ -208,10 +208,37 @@ public int getRunMPwithoutMASC(boolean gravity, boolean ignoreheat, boolean igno
         }
         return getWalkMP(gravity, ignoreheat);
     }
-
+    
+    /*
+     * Normally Quads can't torso twist. Extended torso twist allows regular bipedal mech twisting
+     */
     @Override
     public boolean canChangeSecondaryFacing() {
-        return false;
+    	return hasQuirk(OptionsConstants.QUIRK_POS_EXT_TWIST) && !isProne();
+    }
+    
+    @Override
+    public boolean isValidSecondaryFacing(int dir) {
+        int rotate = dir - getFacing();
```

Comment:
You left out the call to Math.abs, which will result in it not working properly when rotating anticlockwise. Since this has been sitting here idle while I've been away I went ahead and fixed this so we can get it merged.